cranelift-codegen/src/regalloc/liveness.rs

```diff
@@ -448,6 +448,43 @@ impl Liveness {
                         lr.affinity.merge(constraint, &reginfo);
                     }
                 }
+
+                if let Some(sig) = func.dfg.call_signature(inst) {
+                    let sig = &func.dfg.signatures[sig];
+                    let args = func.dfg.inst_variable_args(inst);
+                    if sig.params.len() != args.len() {
+                        panic!(
+                            "Assumption violated: args count == sig.params count ({} != {})",
+                            args.len(),
+                            sig.params.len()
+                        );
+                    }
```

Comment:
This `if` with `panic!` could be replaced by `assert_eq!`, which has the nice property of handling the `({} != {})` part automatically.